### PR TITLE
METRON-2247 - rpm-docker: Provide an option to bypass running rpmlint

### PR DESCRIPTION
**Pull request by tigerquoll**
## Contributor Comments
This PR allows full dev deploy users to skip rpmlint processing via setting an environment variable.

For justification for excluding rpmlint from the build please see https://issues.apache.org/jira/browse/METRON-2247

The PR contains the needed logic to tunnel an option variable from vagrant->ansible->docker to the shell script that actually runs rpmlint

The shell script has been modified to skip running rpmlint if the SKIP_RPMLINT environment variable has been sent.

To demonstrate this option for a single vagrant operation use:

```
env ANSIBLE_ARGS='--extra-vars "SKIP_RPMLINT=1"' vagrant up
```

To use the option permanently place the ANSIBLE_ARGS environment variable in your profile.

Testing:
an exit 1 was added to the build script where the rpmlint command was being run. This causes the build process to die if rpmlint is being run. Running vagrant up without setting the `SKIP_RPMLINT` environment variable causes the build process to die.  Setting the variable  allows a full dev build to continue.

A full dev build was done both with and without `SKIP_RPMLINT` being set to verify the Metron deploys fine either way.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- n/a Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- n/a Have you written or updated unit tests and or integration tests to verify your changes?

- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- n/a Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

- [x] Have you ensured that any documentation diagrams have been updated, along with their source files, using [draw.io](https://www.draw.io/)? See [Metron Development Guidelines](https://cwiki.apache.org/confluence/display/METRON/Development+Guidelines) for instructions.

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by mmiklavc** [comment]
@tigerquoll please update the PR title - the git elves won't match on mixed-case for the Jira logging. Needs to be "METRON-2247 rpm-docker: Provide an option to bypass running rpmlint." As a heads up, github seems to trample the commit message like this even if you set it correctly when committing your changes locally, so you have to manually edit before submitting the PR When you've done that, please close/reopen the PR - hopefully the gitbot will start logging the PR comments ongoing.

File: metron-deployment/development/centos7/ansible.cfg [hunk 1]
```diff
@@ -21,6 +21,7 @@ roles_path = ../../ansible/roles
 pipelining = True
 log_path = ./ansible.log
 callback_plugins = ../../ansible/callback_plugins
+callback_whitelist = profile_tasks
```
[the remark this answers is not in the document]
Comment (nickwallen) [reply metron-deployment/development/centos7/ansible.cfg]: Ah, so that's where the timings were coming from.  Good addition.  I quite like that.

**Comment by nickwallen** [comment]
Thanks for the work @tigerquoll .  I think this is a useful addition. +1

Let's try to get +0 or better from those that commented on the JIRA before merging.... @justinleet and @mmiklavc.

**Comment by mmiklavc** [comment]
Nice @tigerquoll! +1 from me.

**Comment by justinleet** [comment]
+1, this is a great quality of life feature

**Pull request closed without merging**